src/gpodder/gtkui/main.py

```diff
@@ -3695,6 +3695,26 @@ def extensions_episode_download_cb(self, episode):
         logger.debug('extension_episode_download_cb(%s)', episode)
         self.download_episode_list(episodes=[episode])
 
+    def mount_volume_cb(self, file, res, mount_result):
+        result = True
+        try:
+            file.mount_enclosing_volume_finish(res)
+        except GLib.Error as err:
+            if (not err.matches(Gio.io_error_quark(), Gio.IOErrorEnum.NOT_SUPPORTED) and
+                not err.matches(Gio.io_error_quark(), Gio.IOErrorEnum.ALREADY_MOUNTED)):
+                logger.error('mounting volume %s failed: %s' % (file.get_uri(), err.message));
+                result = False
+        finally:
+            mount_result["result"] = result
+            Gtk.main_quit()
+
+    def mount_volume_for_file(self, file):
+        mount_result = {}
+        op = Gtk.MountOperation.new(self.main_window)
+        file.mount_enclosing_volume(Gio.MountMountFlags.NONE, op, None, self.mount_volume_cb, mount_result)
+        Gtk.main()
```

Comment:
hum. This should only be called on program entry,  right ?

Reply:
As I understood it, it's ok to nest these calls, from https://developer.gnome.org/gtk3/stable/gtk3-General.html:
```
void
gtk_main (void);

Runs the main loop until gtk_main_quit() is called.

You can nest calls to gtk_main(). In that case gtk_main_quit() will make the innermost invocation of the main loop return.
```
I am using this because there's no synchronous version of mount_enclosing_volume. (The alternative approach here would be to make device opening async too).

Reply:
wrt to Windows and MacOS I don't know how gio works -- I will check in Windows but I don't have access to a Mac

Reply:
> I am using this because there's no synchronous version of mount_enclosing_volume. (The alternative approach here would be to make device opening async too).

How does one `Gtk.main_quit()` match with the `Gtk.main()`? OK, there seems to be `Gtk.main_level()`, etc.. I don't know enough to be certain but it looks good.